### PR TITLE
[CIVIS-5620] Ruby 3 support

### PR DESCRIPTION
**Pull request by vfu116**
Upgrade swagger-diff to be compatible with Ruby 3. Most of the changes here are linting or gem upgrade related

### CODE REVIEW

File: swagger-diff.gemspec [hunk 1]
```diff
@@ -25,15 +25,16 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.required_ruby_version = '~> 2.0'
+  spec.required_ruby_version = '>= 2.7'
 
-  spec.add_dependency 'json-schema', '~> 2.6'
+  spec.add_dependency 'json-schema', '~> 3.0'
```
Comment (vfu116) [inline swagger-diff.gemspec:30]: ruby 3 support theoretically beings at this version https://github.com/voxpupuli/json-schema/blob/master/CHANGELOG.md

**Comment by vfu116** [comment]
fiddled around with the updated gem in console tests and things look good there:
https://github.com/civisanalytics/console/pull/16600

File: swagger-diff.gemspec [hunk 2]
```diff
@@ -25,15 +25,16 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.required_ruby_version = '~> 2.0'
+  spec.required_ruby_version = '>= 2.7'
```
Comment (anja-schulz) [inline swagger-diff.gemspec:28]: bumping to 2.7 makes sense to me!

**Review by anja-schulz — APPROVED** [review]
lgtm if it all looked good in staging too!!

**Pull request merged**